assets/src/components/activities/adaptive/components/authoring/LayoutEditor.tsx

```diff
@@ -37,97 +43,86 @@ const defaultHandler = async () => {
-  const [portalId, setPortalId] = useState(props.configurePortalId || fallbackPortalId);
-
-  useEffect(() => {
-    setPortalId(props.configurePortalId || fallbackPortalId);
-  }, [props.configurePortalId]);
+  const portalId = props.configurePortalId || fallbackPortalId;
 
   // this effect keeps the local parts state in sync with the props
   useEffect(() => {
     setParts(props.parts);
   }, [props.parts]);
 
-  // this effect keeps the local selected id in sync with the props
-  useEffect(() => {
-    if (props.selected !== selectedPartId) {
-      setSelectedPartId(props.selected || '');
-    }
-  }, [props.selected]);
-
-  // this effect keeps the toolbar positioned next to the selected part
-  useEffect(() => {
-    const x = selectedPart?.custom.x || 0;
-    const y = (selectedPart?.custom.y || 0) + toolBarTopOffset;
-    if (toolbarPosition.x !== x && toolbarPosition.y !== y) {
-      setToolbarPosition({ x, y });
-    }
-  }, [selectedPart]);
-
-  // this keeps a reference to the actual part data of the selected part id in local state
-  useEffect(() => {
-    if (selectedPartId) {
-      const part = parts.find((p) => p.id === selectedPartId);
-      if (part) {
-        let capabilities = { ...defaultCapabilities };
-        // attempt to get an instance of the part class
-        const PartClass = customElements.get(part.type);
-        if (PartClass) {
-          // TODO: cache the instance data somewhere so we don't do this every time
-          const instance = new PartClass() as any; // TODO: extend HTMLElement?
-          if (instance.getCapabilities) {
-            capabilities = { ...capabilities, ...instance.getCapabilities() };
-          }
-          const partWithCapabilities = { ...part, capabilities };
-          setSelectedPart(partWithCapabilities);
-          /* console.log('PART SELECTION CHANGED', {
-            selectedPartId,
-            selectedPart: partWithCapabilities,
-          }); */
-        }
-      }
-    } else {
-      setSelectedPart(null);
-      /* console.log('PART SELECTION CHANGED', {
-        selectedPartId,
-        selectedPart: null,
-      }); */
-    }
-    // any time selection changes we need to stop editing
-    setConfigurePartId('');
-  }, [selectedPartId, parts]);
+  const toolbarPosition = { x: 0, y: 0 };
+  if (selectedPartAndCapabilities) {
+    const x = selectedPartAndCapabilities?.custom.x || 0;
+    const y = (selectedPartAndCapabilities?.custom.y || 0) + toolBarTopOffset;
+    toolbarPosition.x = x;
+    toolbarPosition.y = y;
+  }
 
   // this effect is to cover the case when the user is clicking "off" of a part to deselect it
   useEffect(() => {
     const handleHostClick = (e: any) => {
       const path = e.path || (e.composedPath && e.composedPath());
       const pathIds =
         path?.map((node: HTMLElement) => node.getAttribute && node.getAttribute('id')) || [];
+      const classes = path?.map((node: HTMLElement) => node.className) || [];
```

Comment:
After replacing react-draggable with react-rnd, I ended up running into an odd case when trying to select a component with a shadow-dom. This check for isDraggable fixed that.